subsys/net/ip/dhcpv4.c

```diff
@@ -1006,6 +1006,11 @@ static enum net_verdict net_dhcpv4_input(struct net_conn *conn,
 		return NET_DROP;
 	}
 
+	if (msg->hlen != HARDWARE_ETHERNET_LEN) {
```

Comment:
Currently we only have Ethernet based networks that use DHCP so this works ok at the moment.
But I think the proper check should be
`if (msg->hlen != net_if_get_link_addr(iface)->len)) {`
so checking against the real length of the hardware address. If we do it like this, we do not need to change this in the future if the hw length is different than 6.

Reply:
We should change the hlen when creating the dhcpv4 packet. So in `dhcpv4_create_message()` we need to have
`msg->hlen = net_if_get_link_addr(iface)->len;` instead of
`msg->hlen  = HARDWARE_ETHERNET_LEN;`

Reply:
Thank you for explanation, I fixed it according to your notes